frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -280,7 +262,7 @@ fn decl_outer_config<'a>(
 			})
 		})
 		.map(|(pallet_declaration, generics)| {
-			let pallet = &pallet_declaration.pallet;
+			let pallet = &pallet_declaration.pallet.inner.segments.last().unwrap();
```

Comment:
The problem is with the declarative macros, we can't actually use the full path, since it only accepts an identifier, hence why I'm only pulling the last path segment out to preserve the old behaviour.

Reply:
#8862

Reply:
maybe we should concatenate all the path segments with `_` thus it should preserve past behavior and be unique.

Reply:
I know this doesn't prevent 2 names from being the same, but I think it is better than just using last segment. 

Maybe we should do https://github.com/paritytech/substrate/issues/8862 in this PR too

Reply:
For now `decl_outer_config` has been migrated to become a proc macro, and the enum variants are using the full path snake case naming scheme.